### PR TITLE
ipopt: add 3.14.12 to 3.14.14, improve mumps integration

### PR DESCRIPTION
**Pull request by chapman39**
* Adds new ipopt versions
* Resolves issues using mumps as linear solver in an parallel program
 
Using mumps as ipopt's solver requires building `mumps+mpi` and setting `--disable-mpiinit` in ipopt's configuration. Additional information can be seen in this Ipopt PR https://github.com/coin-or/Ipopt/pull/790

### CODE REVIEW

**Comment by chapman39** [comment]
@spackbot fix style

**Comment by chapman39** [comment]
@goxberry Is no longer an active maintainer of the Ipopt Spack package.

**Comment by goxberry** [comment]
I can't test these changes myself, but as far as I can see, the changes look fine. I haven't used IPOPT since leaving LLNL, and my bandwidth for Spack reviews is limited, so I support the change in maintainers.

**Review by goxberry — APPROVED** [review]

File: var/spack/repos/builtin/packages/ipopt/package.py [hunk 1]
```diff
@@ -112,8 +118,10 @@ def configure_args(self):
                         "--with-mumps-cflags=%s" % mumps_dir.include,
                     ]
                 )
+                if "^mumps+mpi" in spec:
+                    args.extend(["--disable-mpiinit"])
 
-        if spec.satisfies("^coinhsl"):
+        if spec.satisfies("+coinhsl"):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ipopt/package.py:124]: Great catch!

File: var/spack/repos/builtin/packages/ipopt/package.py [hunk 2]
```diff
@@ -144,6 +156,11 @@ def configure_args(self):
                     ]
                 )
 
+        if spec.satisfies("+java"):
+            args.extend(["--enable-java"])
+        else:
+            args.extend(["--disable-java"])
+
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ipopt/package.py:163]: You could simplify this using the `enable_or_disable` helper (see docs at https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#addings-flags-to-configure).

**Review by tldahlgren — APPROVED** [review]
Confirmed the three new version sha256s.  Let me know if you want to simplify the `java` variant option handling.

**Review by bernhardkaindl — APPROVED** [review]
Build results on Ubuntu 22.04.4 LTS:
```py
Passed: ipopt@3.14.14 ipopt@3.14.13 ipopt@3.14.12 ipopt+java
spack find -v:
-- linux-ubuntu22.04-skylake / gcc@13.1.0 -----------------------
ipopt@3.14.12~coinhsl~debug~java~metis+mumps build_system=autotools patches=cc52643
ipopt@3.14.13~coinhsl~debug~java~metis+mumps build_system=autotools patches=cc52643
ipopt@3.14.14~coinhsl~debug~java~metis+mumps build_system=autotools patches=cc52643
ipopt@3.14.14~coinhsl~debug+java~metis+mumps build_system=autotools patches=cc52643
```

File: var/spack/repos/builtin/packages/ipopt/package.py (hunk 2 above)
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/ipopt/package.py:163]: @chapman39: Tamara's suggestion is good, try it with the next update, but this should be fine.

As the change to `enable_or_disable` would need a new CI build, and it is only one use of it, not many dozens of it, I hope it's not a problem to merge and not require a revision and rebuild.

Comment (chapman39) [reply var/spack/repos/builtin/packages/ipopt/package.py:163]: good to know, thanks!

**Pull request merged**